### PR TITLE
image-builder: check /sbin/init when AGENT_INIT=yes

### LINKED ISSUE #322: Unable to build image: missing kata-agent binary
Creating image fails because of missing kata-agent binary.

```
$ sudo -E PATH="${PATH}" make USE_DOCKER=true AGENT_INIT=yes DISTRO=ubuntu rootfs
...<snip>...
INFO: Pull Agent source code
[OK] Pull Agent source code
INFO: Build agent
/go/src/github.com/kata-containers/agent /
rm -f kata-agent kata-agent.service
go build -buildmode=pie -tags "" -o kata-agent \
	-ldflags "-X main.version=1.8.0-alpha2-bad7d1b151295af9620d3643cb9e43cef97777fd -X main.seccompSupport=no "
install -D kata-agent /rootfs/usr/bin/kata-agent
/
[OK] Agent installed
INFO: Install /rootfs/usr/bin/kata-agent as init process
[OK] Agent is installed as init process
INFO: Check init is installed
[OK] init is installed
INFO: Creating summary file
INFO: Created summary file '/var/lib/osbuilder/osbuilder.yaml' inside rootfs
touch /home/vagrant/osbuilder/.ubuntu_rootfs.done


$ sudo -E PATH="${PATH}" make USE_DOCKER=true AGENT_INIT=yes DISTRO=ubuntu image
Creating image based on /home/vagrant/osbuilder/ubuntu_rootfs
/home/vagrant/osbuilder/image-builder/image_builder.sh "/home/vagrant/osbuilder/ubuntu_rootfs"
Sending build context to Docker daemon  45.57kB
Step 1/3 : From fedora:latest
 ---> 289289d1a15b
Step 2/3 : RUN [ -n "$http_proxy" ] && sed -i '$ a proxy='$http_proxy /etc/dnf/dnf.conf ; true
 ---> Using cache
 ---> a4a76abf081a
Step 3/3 : RUN dnf install -y qemu-img parted gdisk e2fsprogs gcc xfsprogs findutils
 ---> Using cache
 ---> ebfebe18888d
Successfully built ebfebe18888d
Successfully tagged image-builder-osbuilder:latest
[OK] init is installed
ERROR: /usr/bin/kata-agent is not installed in /rootfs. Use AGENT_BIN env variable to change the expected agent binary name
ERROR: Invalid rootfs
Makefile:82: recipe for target '/home/vagrant/osbuilder/kata-containers.img' failed
make: *** [/home/vagrant/osbuilder/kata-containers.img] Error 1
```

The agent was actually [moved](https://github.com/kata-containers/osbuilder/blob/a84fcf296db88560cfe52bf501fdad6996171766/rootfs-builder/rootfs.sh#L217) to `/sbin/init` by rootfs.sh.
```
$ ls -lh ubuntu_rootfs/sbin/init
-rwxr-xr-x 1 root root 25M Jun 25 05:20 ubuntu_rootfs/sbin/init
```

### PR DESCRIPTION
**Pull request by devimc**
kata-agent is moved to /sbin/init when AGENT_INIT is yes. Check
/sbin/init exists and it's not systemd

fixes #322

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

File: image-builder/image_builder.sh [hunk 1]
```diff
@@ -193,12 +194,18 @@ check_rootfs() {
...
 		"yes")
-			agent_path="/usr/bin/${AGENT_BIN}"
+			agent_path="/sbin/init"
 			agent="${rootfs}${agent_path}"
 			if  [ ! -x "${agent}" ]; then
 				error "${agent_path} is not installed in ${rootfs}. Use AGENT_BIN env variable to change the expected agent binary name"
 				return 1
 			fi
+			# checksum must be different to system
+			if [ -f "${systemd}" ] && [ "$(cmp -s "${systemd}" "${agent}")" ]; then
```
Comment (jodh-intel) [inline image-builder/image_builder.sh]: That `cmp` doesn't look right to me - you're checking the *output* of `cmp`, but running with `-s` meaning "no output"... ? Can't it be:

```sh
if [ -f "${systemd}" ] && cmp -s "${systemd}" "${agent}"; then
```

Comment (devimc) [reply image-builder/image_builder.sh]: good catch! you're right, thanks

**Comment by devimc** [comment]
/test

@jodh-intel change applied, thanks

**Comment by jodh-intel** [comment]
More reviews required over here folk...

**Pull request merged**